dpnp/backend/extensions/lapack/gesv.cpp

```diff
@@ -69,12 +64,32 @@ static sycl::event gesv_impl(sycl::queue &exec_q,
     const std::int64_t lda = std::max<size_t>(1UL, n);
     const std::int64_t ldb = std::max<size_t>(1UL, n);
 
-    const std::int64_t scratchpad_size =
+    std::int64_t scratchpad_size = 0;
+    sycl::event comp_event;
+    std::int64_t *ipiv = nullptr;
+    T *scratchpad = nullptr;
+
+    std::stringstream error_msg;
+    bool is_exception_caught = false;
+
+#if defined(USE_ONEMKL_INTERFACES)
+    // Use transpose::T if the LU-factorized array is passed as C-contiguous.
+    // For F-contiguous we use transpose::N.
+    // Since gesv takes F-contiguous as input, we use transpose::N.
+    oneapi::mkl::transpose trans = oneapi::mkl::transpose::N;
+
+    scratchpad_size = std::max(
+        mkl_lapack::getrf_scratchpad_size<T>(exec_q, n, n, lda),
+        mkl_lapack::getrs_scratchpad_size<T>(exec_q, trans, n, nrhs, lda, ldb));
+
+#else
+    scratchpad_size =
         mkl_lapack::gesv_scratchpad_size<T>(exec_q, n, nrhs, lda, ldb);
 
-    T *scratchpad = helper::alloc_scratchpad<T>(scratchpad_size, exec_q);
+#endif // USE_ONEMKL_INTERFACES
+
+    scratchpad = helper::alloc_scratchpad<T>(scratchpad_size, exec_q);
```

Comment:
no need to have a declaration at line#70
```suggestion
    T *scratchpad = helper::alloc_scratchpad<T>(scratchpad_size, exec_q);
```